### DISCUSSION TITLE
How to avoid exhausted renderings?

### DISCUSSION THREAD

**Discussion by wuarmin** [table-library/react-table-library#23] [Q&A]
Hey,

I observe unnecessary renderings of the Table-Body. Is there a solution to avoid this. For example, if a state changes that is only consumed by the HeaderRow, the entire Body is rerendered, because the table recreates the modifiedNodes on every render.

Do you have any hint?

Best regards,
thanks

**Comment by rwieruch** [comment]
Hi @wuarmin I didn't do much performance enhancements yet. In your case, if `modifiedNodes` is causing a re-render, maybe we should try to add a `useMemo` for it in a PR. Would you be able to test whether this helps your case?

**Reply by wuarmin** [reply]
I will try it, if I find some time. 👍
